fiona/drvsupport.py

```diff
@@ -191,12 +223,67 @@ def _filter_supported_drivers():
         for drv in supported_drivers.keys():
             if drv not in ogrdrv_names:
                 del supported_drivers_copy[drv]
+                continue
 
     supported_drivers = supported_drivers_copy
 
 
 _filter_supported_drivers()
 
+
+@require_gdal_version("2.0")
+def vector_driver_extensions():
+    """
+    Returns
+    -------
+    dict:
+        Map of extensions to the driver.
+    """
+    from fiona.meta import extensions  # prevent circular import
+
+    extension_to_driver = _extension_to_driver.copy()
+    for drv, modes in supported_drivers.items():
+        # update extensions based on driver suppport
+        for extension in extensions(drv) or ():
+            if "w" in modes:
```

Comment:
I figured this would be better than checking if GDAL supports it so that a better error message will be raised if a driver is selected that is not supported downstream.